spec/acceptance/class_spec.rb

```diff
@@ -13,14 +13,18 @@
   service_name          = 'fail2ban'
   ssh_log_file          = '/var/log/secure'
   ssh_jail              = 'ssh'
-  # EPEL needs to be installed, otherwise it won't work
-  shell('wget http://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm')
-  shell('rpm -ivh epel-release-latest-7.noarch.rpm && yum -y install redhat-lsb-core')
 end
 
 # Ensure the ssh log file is created, otherwise the service doesn't start completely
 shell("touch #{ssh_log_file}")
 
+# CentOS6 is really picky, puppet/beaker doesn't see the fact correctly, which is why it's better left as the "default"
+# It's an ugly hack but it works.
+distname = fact('lsbdistcodename')
```

Comment:
This fact should be present if you install `redhat-lsb-core`. Doing these kind of things outside of describe or other blocks really screws up the order and this can be done before the spec helper is done running. That could mean it's applying this to CentOS 7 as well.